client/client.go

```diff
@@ -442,9 +443,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		return nil, nil, err
 	}
 
+	isWait := false
+	if req != nil && req.URL != nil {
+		isWait, _ = strconv.ParseBool(req.URL.Query().Get("wait"))
+	}
+
 	var hctx context.Context
 	var hcancel context.CancelFunc
-	if c.headerTimeout > 0 {
+	if !isWait && c.headerTimeout > 0 {
```

Comment:
Please check the comment here https://github.com/coreos/etcd/blob/master/client/client.go#L123-L124.

Basically, the requestHeader behavior is undefined when there is a proxy in front of the client. 

You are changing the default behavior from applying the timeout to not apply it. This change seems to be OK with me. Even if we "fix" etcd proxy, we will not be able to control all the proxies people are using. 

But we should make the document better by saying for wait request we will ignore this timeout.